src/physics/em/detail/LPMCalculator.hh

```diff
@@ -0,0 +1,249 @@
+
+  public:
+    // Construct with material data and photon energy
+    inline CELER_FUNCTION LPMCalculator(const MaterialView& material,
+                                        const ElementView&  element,
+                                        bool dielectric_suppression,
+                                        units::MevEnergy gamma_energy);
+
+    // Compute the LPM supression functions
+    inline CELER_FUNCTION LPMFunctions operator()(real_type epsilon);
+
+  private:
+    //// DATA ////
+
+    // Current element
+    const ElementView& element_;
+    // Electron density of the current material [1/cm^3]
+    const real_type electron_density_;
+    // Characteristic energy for the LPM effect for this material [MeV]
+    const real_type lpm_energy_;
+    // Include a dielectric suppression effect
+    const bool dielectric_suppression_;
+    // Photon energy [MeV]
+    const real_type gamma_energy_;
+
+    //// HELPER FUNCTIONS ////
+
+    inline CELER_FUNCTION LPMFunctions compute_g_phi(real_type s) const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with LPM data, material data, and photon energy.
+ */
+CELER_FUNCTION
+LPMCalculator::LPMCalculator(const MaterialView& material,
+                             const ElementView&  element,
+                             bool                dielectric_suppression,
+                             units::MevEnergy    gamma_energy)
+    : element_(element)
+    , electron_density_(material.electron_density())
+    , lpm_energy_(material.radiation_length()
+                  * value_as<MevPerCm>(lpm_constant()))
+    , dielectric_suppression_(dielectric_suppression)
+    , gamma_energy_(gamma_energy.value())
+{
+    CELER_EXPECT(gamma_energy_ > 0);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Compute the LPM suppression functions.
+ *
+ * Here \f$ \epsilon \f$ is the ratio of the  electron (or positron) energy to
+ * the photon energy, \f$ \epsilon = E / k \f$.
+ *
+ * See section 10.2.2 of the Geant4 Physics Reference Manual and
+ * ComputeLPMfunctions and GetLPMFunctions in G4eBremsstrahlungRelModel and
+ * G4PairProductionRelModel.
+ */
+CELER_FUNCTION auto LPMCalculator::operator()(real_type epsilon) -> LPMFunctions
+{
+    // Suppression variable \f$ s' \f$. For bremsstrahlung \f$ s' =
+    // \sqrt{\frac{E_\textrm{LPM} k}{8 E (E - k)}} \f$, and for pair production
+    // \f$ s' = \sqrt{\frac{E_\textrm{LPM} k}{8 E (k - E)}} \f$, where \f$ k \$
+    // is the photon energy and \f$ E \f$ is the electon (or positron) energy
+    const real_type s_prime = std::sqrt(
+        lpm_energy_ / (8 * epsilon * gamma_energy_ * std::fabs(epsilon - 1)));
+
+    // TODO: In the Geant4 relativistic pair production model the denominator
+    // is 184 instead of 184.15 -- why? Will it matter?
```

Comment:
Some references do indeed have 184 (e.g., [Klein](https://doi.org/10.1103/RevModPhys.71.1501)), although the original calculation by [Tsai](https://doi.org/10.1103/RevModPhys.46.815) was 184.15. It looks like this was all based on some numerical approximations and various models, so there shouldn't be any practical difference between 184 and 184.15.